### DISCUSSION TITLE
Time series plot extent

### DISCUSSION THREAD

**Discussion by daniel-caichac-DHI** [DHI/modelskill#437] [General]
Hi, 
short question, but I think it comes from this line
https://github.com/DHI/modelskill/blob/72d51833a284a1b7b3d195a61de1ca8ea139e89b/modelskill/comparison/_comparer_plotter.py#L99

But in a nutshell, when I have a matched model-obs (ie, after doing `ms.match`) and then I do a Scatter plot, I only get overlapping data of course, which is right, but a timeseries plots, is plotting the `raw` model data on the back and then the measurements on the top, so not necessarily the _matched_ time steps.
This is kind of funny when the model is much longer than the measurements, see example (which also makes it slower)
![image](https://github.com/DHI/modelskill/assets/97288080/80ebcf0a-f640-437f-8c9b-065fca7fb478)

I am bypassing this by manually adding `ax.set_xlim` after the plots is done (I know it can also be a `kwarg`), but since the histogram and scatter and all statistics are done on the `paired` data, is there a particular reason why the timeseries plot is done on the raw model and not on the compared data? @ecomodeller  
This is how it looks after manually setting ax.set_xlim
![image](https://github.com/DHI/modelskill/assets/97288080/3cdad975-5601-4d96-aced-4df5e35ea9e4)

**Comment by jsmariegaard** [comment]
The reason for plotting the raw data in the time series plot is that the measurement data can potentially have large gaps or be very sparse. You can can force the modelresult to only cover a specific period with the sel method before matching e.g. mr = mr.sel(time="2023")

**Reply by daniel-caichac-DHI** [reply]
I see. Is just that sometimes a small gap during a peak event in the measurements (say that device was flagged wrong) could mislead the reader/user into thinking that the model has a huge peak that the measurements don't, but is just an artifact of the Model being plotted while the measurement is not, and in that case the `.sel` will not fix it, see  eg April/2023 in the picture I just plotted (copy/pasted below)
![image](https://github.com/DHI/modelskill/assets/97288080/ea6dfe40-27fe-4e04-aed8-6a1c45268a45)
Since the timeseries plot is the easier to reproduce I will just plot on my own the model and measurements from the `comparer` itself, where times have been paired already, but I still wanted to raise this point in case you think is worth modifying (or not).

**Reply by ecomodeller** [reply]
Some measurements are done by manual sampling, and the sampling frequency can be weeks or months, and then it is very useful to be able to see the model dynamics at all timesteps, and not limit the plot to when times when the measurements are taken.

**Reply by daniel-caichac-DHI** [reply]
Yes I see, both approaches have their own pro/cons. 

I think then, that we should just leave as it is, since it is quite easy for the user to just plot directly from the comparer if only co-existing timesteps are wanted.
Thanks

**Reply by jsmariegaard** [reply]
Alternatively, we could add an argument `plot_matched_only` ...
